### DISCUSSION TITLE
iguana支持匿名类么？该如何写

### DISCUSSION THREAD

**Discussion by 171930433** [alibaba/yalantinglibs#645] [Q&A] [answered]
```
struct AS
{
struct IAS{
double y;
};

REFLECTION(IAS,y);

double x;
IAS x2;
};

REFLECTION(AS,x,x2);

int main() {

  AS a;
  std::string str;
  iguana::to_json(a,str);

  std::cout << str <<"\n";

  return 0;
}
```

编译失败：
no matching function for call to ‘to_json_impl<true>(std::__cxx11::basic_string<char>&, AS::IAS&)’

**Comment by poor-circle** [comment]
Could you try:
```cpp
struct AS
{
struct IAS{
double y;
};

double x;
IAS x2;
};

REFLECTION(AS::IAS,y);
REFLECTION(AS,x,x2);

int main() {

  AS a;
  std::string str;
  iguana::to_json(a,str);

  std::cout << str <<"\n";

  return 0;
}
```

**Reply by 171930433** [reply]
编译失败
/home/gsk/pro/yalantinglibs/src/example/iguana/reflection.hpp:578:57: error: ‘arr_AS’ has not been declared
  578 |   static constexpr inline std::array<frozen::string, N> arr_##STRUCT_NAME = { \
      |                                                         ^~~~
/home/gsk/pro/yalantinglibs/src/example/iguana/reflection.hpp:578:57: note: in definition of macro ‘MAKE_META_DATA’
  578 |   static constexpr inline std::array<frozen::string, N> arr_##STRUCT_NAME = { \  等

**Comment by qicosmos** [comment, marked as answer]
这应该是一个bug，你先用这个方法绕过去，我晚点修复一下这个bug
```c++
struct AS {
  struct IAS {
    double y;
  };

  double x;
  IAS x2;
};

using IAS = AS::IAS;
REFLECTION(IAS, y);
REFLECTION(AS, x, x2);
```
